src/helpers.rs

```diff
@@ -291,4 +291,17 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    /// Helper function to get a `libc` constant as a `Scalar`.
+    fn eval_libc(&mut self, name: &str) -> InterpResult<'tcx, Scalar<Tag>> {
+        self.eval_context_mut()
+            .eval_path_scalar(&["libc", name])?
+            .ok_or_else(|| err_unsup_format!("Path libc::{} cannot be resolved.", name).into())
+            .and_then(|scalar| scalar.not_undef())
```

Comment:
We usually use `?` instead of `and_then`. IMO it makes sense to stick to a consistent style.

Reply:
Oh now I remember, that was because the type inference gets funny there, so to be able to just `?` we would need to do something like:
```rust
        let scalar: InterpResult<'tcx, ScalarMaybeUndef<Tag>> = self.eval_context_mut()
            .eval_path_scalar(&["libc", name])?
            .ok_or_else(|| err_unsup_format!("Path libc::{} cannot be resolved.", name).into());
        scalar?.not_undef()
```

Reply:
nvm: turbofish to the rescue

Reply:
I dont think you need the .into with that, as ? does that implicitely for the error value. That would also probably fix the type inference problem.

Reply:
Yes you're right, I'm going to amend this, there are enough commits here